build/go.bzl

```diff
@@ -0,0 +1,103 @@
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_test")
+
+# Defines several go_binary rules to work around a Bazel issue which makes
+# the pure attribute on go_binary not configurable.
+# The name provided will have cgo enabled if targeting Linux, and will
+# be a pure go binary otherwise. Additionally, if targeting Windows, the
+# output filename will have a .exe suffix.
+def go_binary_conditional_pure(name, tags = None, **kwargs):
```

Comment:
this also "broke" the output locations AFAICT, 

`bazel-bin/cmd/kubelet/linux_amd64_stripped/kubelet` -> `bazel-bin/cmd/kubelet/kubelet`

`bazel-bin/cmd/kubelet/linux_amd64_stripped` has this now `bazel-bin/cmd/kubelet/linux_amd64_stripped/_kubelet-cgo%/k8s.io/kubernetes/cmd/kubelet.a`

Reply:
actually only for kubelet.

I worked around this in https://github.com/kubernetes-sigs/kind/pull/335, but kubernetes-anywhere jobs are broken by this I think.